client/packages/common/src/hooks/useDialog/useDialog.tsx

```diff
@@ -239,7 +239,7 @@ export const useDialog = (dialogProps?: DialogProps): DialogState => {
                 <div>{slideConfig.in && children}</div>
               </Slide>
             ) : (
-              <div>{children}</div>
```

Comment:
Why was this removed? I'm all for removing excessive divs but pretty nervous about this one, we use this hook everywhere and wouldn't be surprised if some modal is relying on it for layout?

Reply:
I'm on a mission to hunt down and kill spurious `<div>`s. (Numerous layout bugs have been caused by them)

I'm 90% sure this does absolutely nothing (no styling, or React component), but I'll check the implementations of `useDialog` just to be sure (all the children should have their own top-most structural element).

The only way it could have any effect is if the top-most container of the `children` is an `inline` element rather than a `block` element, but if that was the case I'd rather fix that case than have redundant block containers here.

Will report back shortly.

Reply:
I'll also check the "Slide" case above and remove that div if it's not doing anything.

(Sorry for polluting the PR, I just got frustrated trying to inspect the centering layout of the modal content, and came across this redundant <div> in the HTML)

Reply:
![Screenshot 2024-06-04 at 3 43 58 PM](https://github.com/msupply-foundation/open-msupply/assets/5456533/91c4352b-3c8c-496b-8686-d950ddde8e8b)
<img width="434" alt="Screenshot 2024-06-04 at 3 48 41 PM" src="https://github.com/msupply-foundation/open-msupply/assets/5456533/bbda1ce2-133d-4f6f-bb89-b3b55f6109e3">

Yeah, the `DialogContent` component already renders a `div`, so we end up with `<div><div>{content}</div></div>`, which is pretty sloppy web development 😉.

*However*... the one inside the Slide component does actually seem to be doing something -- seems to embed some CSS classes on the inner div to create the slide animation, and throws and error when missing. WHich is wierd, I'm not sure why it would just target the next top-most block inside it. But in light of that, I'll leave the other div in place now for consistency. (Something to look into another day)

da2af02d5